tasmota/xdrv_10_rules.ino

```diff
@@ -471,11 +472,19 @@ bool RulesRuleMatch(uint8_t rule_set, String &event, String &rule, bool stop_all
       rule_param = String(SunMinutes(1));
     }
 #endif  // USE_TIMERS and USE_SUNRISE
+#if defined(USE_TIMERS)
+    for (uint32_t i = 0; i < MAX_TIMERS; i++) {
+      snprintf_P(stemp, sizeof(stemp), PSTR("%%TIMER%d%%"), i +1);
+      if (rule_param.startsWith(stemp)) {
+        rule_param = String(Settings->timer[i].time);
+        break;
+      }
+    }
```

Comment:
Despite the PR was close, if you want to re-open it again, you should optimize this.
Comparing with 16 possible substrings that won't be there most of the time is CPU consuming
You may search 1st for `%timer` and only if found, look for a terminating number+`%`